### PR TITLE
Fix pervasive warnings on z/OS and AIX

### PR DESCRIPTION
**Pull request by fjeremic**
The following warning is emitted when compiling a compilation unit
which includes the OMRCodeCache.hpp file:

```
"omr/compiler/runtime/OMRCodeCache.hpp", line 450.14: CCN8891 (W) The
array member "void *_CCPreLoadedCode[]" may only be followed by members
of consistent type.
```

The reason this warning is emitted is because `TR_numCCPreLoadedCode`
evaluates to 0 on S390 due to an ifdef and so we have a zero length
array on z/OS. It is emitting a warning because we have types declared
after the zero length array which are not of the same type as the array
elements.

To fix this issue we simply omit an ifdef which forces the value of
`TR_numCCPreLoadedCode` to be non-zero on S390. This is reasonable, as
this enum is not used at all in the S390 codegen.

When a compilation unit including the OMRMonitor.hpp is compiled xlC
will emit the following warnings:

```
"omr/compiler/infra/OMRMonitor.hpp", line 50.46: CCN6101 (W) A return
value of type "int" is expected.
"omr/compiler/infra/OMRMonitor.hpp", line 54.77: CCN6101 (W) A return
value of type "long" is expected.
"omr/compiler/infra/OMRMonitor.hpp", line 57.48: CCN6101 (W) A return
value of type "int" is expected.
```

It appears that xlC is unable to deduce that the function will not
return and as such it emits a warning saying that there is no return
value. To fix this we just return default values. It doesn't matter
what we return really because at runtime we will never execute the
return statement.

Fixes: #3703
Fixes: #4548

### CODE REVIEW

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by fjeremic** [comment]
@genie-omr build zos

**Comment by fjeremic** [comment]
@genie-omr build zos

**Comment by fjeremic** [comment]
@genie-omr build zos

**Comment by fjeremic** [comment]
Examples of pervasive warnings before this change:
https://ci.eclipse.org/omr/job/PullRequest-zos_390-64/2250/console

Example after this change:
https://ci.eclipse.org/omr/job/PullRequest-zos_390-64/2255/console

File: compiler/infra/OMRMonitor.hpp [hunk 1]
```diff
@@ -47,14 +47,14 @@ class Monitor
    static TR::Monitor *create(char *name);
    static void destroy(TR::Monitor *monitor);
    void enter();
-   int32_t try_enter() { TR_UNIMPLEMENTED(); }
+   int32_t try_enter() { TR_UNIMPLEMENTED(); return 0; }
    int32_t exit(); // returns 0 on success
    void destroy();
    void wait() { TR_UNIMPLEMENTED(); }
-   intptr_t wait_timed(int64_t millis, int32_t nanos) { TR_UNIMPLEMENTED(); }
+   intptr_t wait_timed(int64_t millis, int32_t nanos) { TR_UNIMPLEMENTED(); return NULL; }
```
Comment (0xdaryl) [inline compiler/infra/OMRMonitor.hpp:54]: Returning `NULL` here makes it seem like this function is returning a pointer.  Although the API for this function does not have any documentation in OMR, its use in downstream projects seems to suggest that this function returns a status code.

Can you return `0` instead?

Comment (fjeremic) [reply compiler/infra/OMRMonitor.hpp]: Fixed in [force-push](https://github.com/eclipse/omr/compare/dcec147dabf39419d0f87cf6f678d1ed2e9b6224..30a19e4b8c3a65c189c174ccdb3970ca868203d9).

**Comment by 0xdaryl** [comment]
Recent change is functionally equivalent to the previous commit for which full CI was run.  No need to re-run CI.

**Pull request merged**